applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+                auto& r_geometry = it_elem->GetGeometry();
+
+                // Current geometry information
+                const std::size_t local_space_dimension = r_geometry.LocalSpaceDimension();
+                // KRATOS_WATCH(local_space_dimension);
+                const std::size_t number_nodes_element = r_geometry.PointsNumber();
+                // KRATOS_WATCH(number_nodes_element);
+
+                // Resize if needed
+                if (DN_DX.size1() != number_nodes_element || DN_DX.size2() != dimension)
+                    DN_DX.resize(number_nodes_element, dimension);
+                if (N.size() != number_nodes_element)
+                    N.resize(number_nodes_element);
+                if (J0.size1() != dimension || J0.size2() != local_space_dimension)
+                    J0.resize(dimension, local_space_dimension);
+
+                // Build values vectors of the velocity
+                Vector values_x(number_nodes_element);
+                Vector values_y(number_nodes_element);
+                Vector values_z(number_nodes_element);
+                for(int i_node=0; i_node < number_nodes_element; ++i_node){
+                    values_x[i_node] = r_geometry[i_node].FastGetSolutionStepValue(VELOCITY_X);
+                    values_y[i_node] = r_geometry[i_node].FastGetSolutionStepValue(VELOCITY_Y);
+                    values_z[i_node] = r_geometry[i_node].FastGetSolutionStepValue(VELOCITY_Z);
+                }
+
+                // Set integration points
+                const auto& r_integration_method = r_geometry.GetDefaultIntegrationMethod(); // Default is 0
+                const auto& r_integration_points = r_geometry.IntegrationPoints(r_integration_method); // Default is [3 dimensional integration point(0.333333 , 0.333333 , 0), weight = 0.5]
+                const std::size_t number_of_integration_points = r_integration_points.size(); // Default is 1
+
+                // Set containers of the shape functions and the local gradients
+                const Matrix& rNcontainer = r_geometry.ShapeFunctionsValues(r_integration_method); // [1,3]((0.333333,0.333333,0.333333))
+                const auto& rDN_DeContainer = r_geometry.ShapeFunctionsLocalGradients(r_integration_method); // [1]([3,2]((-1,-1),(1,0),(0,1)))
+
+                // Initialize auxiliary local variables
+                double divergence_current = 0;
+                double velocity_seminorm_current = 0;
+
+                // Loop over integration points
+                for ( IndexType point_number = 0; point_number < number_of_integration_points; ++point_number ){
+                    // Getting the shape functions
+                    noalias(N) = row(rNcontainer, point_number); // [3](0.333333,0.333333,0.333333)
+
+                    // Get the jacobians
+                    GeometryUtils::JacobianOnInitialConfiguration(r_geometry, r_integration_points[point_number], J0);
```

Comment:
Wouldn't this be incorrect if this process is used in an ALE framework?

Reply:
I would actually say it is wrong. you should use the normal jacobian in this context

Reply:
Done